File: ios/SpentyAI/SpentyAI/Features/FeatureRequests/FeatureRequestsRepository.swift
import Foundation

// MARK: - Models

struct FeatureRequest: Codable, Identifiable {
    let id: String
    let title: String
    let description: String?
    let category: FeatureRequestCategory
    let status: FeatureRequestStatus
    let votes: Int
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id, title, description, category, status, votes
        case createdAt = "created_at"
    }
}

enum FeatureRequestCategory: String, Codable, CaseIterable {
    case ui = "UI"
    case performance = "Performance"
    case feature = "Feature"
    case integration = "Integration"
    case other = "Other"
}

enum FeatureRequestStatus: String, Codable {
    case pending
    case inProgress = "in_progress"
    case completed
}

struct CreateFeatureRequestBody: Codable {
    let title: String
    let description: String
    let category: String
}

struct VoteResponse: Codable {
    let votes: Int
}

// MARK: - Repository

final class FeatureRequestsRepository {

    static let shared = FeatureRequestsRepository()
    private init() {}

    func getAll() async throws -> [FeatureRequest] {
        try await APIClient.shared.get(APIEndpoints.featureRequests)
    }

    func create(title: String, description: String, category: FeatureRequestCategory) async throws -> FeatureRequest {
        let body = CreateFeatureRequestBody(
            title: title,
            description: description,
            category: category.rawValue
        )
        return try await APIClient.shared.post(APIEndpoints.featureRequests, body: body)
    }

    func vote(id: String) async throws -> VoteResponse {
        try await APIClient.shared.post(APIEndpoints.featureRequestVote(id))
    }
}

File: android-native/app/src/main/java/com/spentyai/app/features/featurerequests/FeatureRequestsRepository.kt
package com.spentyai.app.features.featurerequests

import com.spentyai.app.core.models.FeatureRequest
import com.spentyai.app.core.network.ApiClient
import com.spentyai.app.core.network.ApiResult
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.buildJsonObject
import kotlinx.serialization.json.put

@Serializable
data class VoteResponse(
    val votes: Int = 0
)

enum class FeatureRequestCategory(val value: String, val displayName: String) {
    FEATURE("feature", "Feature"),
    IMPROVEMENT("improvement", "Improvement"),
    INTEGRATION("integration", "Integration"),
    UI("ui", "UI/UX"),
    OTHER("other", "Other")
}

class FeatureRequestsRepository(private val apiClient: ApiClient) {

    suspend fun getAll(): ApiResult<List<FeatureRequest>> {
        return apiClient.safeApiCall { apiClient.endpoints.getFeatureRequests() }
    }

    suspend fun create(
        title: String,
        description: String,
        category: String
    ): ApiResult<FeatureRequest> {
        val body = buildJsonObject {
            put("title", title)
            put("description", description)
            put("category", category)
        }
        return apiClient.safeApiCall { apiClient.endpoints.createFeatureRequest(body) }
    }

    suspend fun vote(id: String): ApiResult<FeatureRequest> {
        return apiClient.safeApiCall { apiClient.endpoints.voteFeatureRequest(id) }
    }
}
